File: app/src/main/java/net/ballmerlabs/uscatterbrain/DeviceProfile.java
package net.ballmerlabs.uscatterbrain;


import java.util.UUID;

/**
 * General device information and settings storage.
 * Used to refer to a device
 */

public class DeviceProfile {

    private UUID luid;

    public enum HardwareServices {
        WIFIP2P, BLUETOOTHCLASSIC,
        BLUETOOTHLE, ULTRASOUND
    }
    private HardwareServices services;


    public DeviceProfile (HardwareServices services, UUID id) {
        this.services = services;
        this.luid = id;
    }

    public void  update(HardwareServices services) {
        this.services = services;
    }

    public HardwareServices getServices() {
        return services;
    }

    public UUID getLUID(){ return this.luid;}

    public void setLUID(UUID id){this.luid = id;}
}

File: app/src/main/java/net/ballmerlabs/uscatterbrain/DeviceProfile.kt
package net.ballmerlabs.uscatterbrain

import java.util.*

/**
 * General device information and settings storage.
 * Used to refer to a device
 */
class DeviceProfile(var services: HardwareServices, var lUID: UUID) {

    enum class HardwareServices {
        WIFIP2P, BLUETOOTHCLASSIC, BLUETOOTHLE, ULTRASOUND
    }

    fun update(services: HardwareServices) {
        this.services = services
    }

}